store/tikv/txn.go

```diff
@@ -185,20 +184,46 @@ func (txn *tikvTxn) Commit(ctx context.Context) error {
 		connID = val.(uint64)
 	}
 	committer, err := newTwoPhaseCommitter(txn, connID)
-	if err != nil {
+	if err != nil || committer == nil {
 		return errors.Trace(err)
 	}
-	if committer == nil {
-		return nil
+
+	defer func() {
+		if err == nil {
```

Comment:
I guess this can be moved into `executeAndWriteFinishBinlog`, so `defer` is unnecessary